set.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+)
+
+const (
+	Undefined int = iota
+	Bool
+	Number
+	String
+	Array
+	Object
+)
+
+type JsonSet struct {
+	// The Consul Key to set.
+	Key string
+	// The value that was passed by the command line.
+	value string
+	// What we expect the value type to be.
+	ExpectedType int
+	// The actual Json Value that is going to be saved to Consul.
+	JsonValue []byte
+}
+
+func (js *JsonSet) setExpectedType(t string) {
+	switch t {
+	case "bool":
+		js.ExpectedType = Bool
+	case "number":
+		js.ExpectedType = Number
+	case "int":
+		js.ExpectedType = Number
+	case "float":
+		js.ExpectedType = Number
+	case "string":
+		js.ExpectedType = String
+	case "array":
+		js.ExpectedType = Array
+	case "object":
+		js.ExpectedType = Object
+	default:
+		js.ExpectedType = Undefined
+	}
+}
+
+func (js *JsonSet) expectedType() string {
+	switch js.ExpectedType {
+	case Bool:
+		return "bool"
+	case Number:
+		return "number"
+	case String:
+		return "string"
+	case Array:
+		return "array"
+	case Object:
+		return "object"
+	}
+
+	return "undefined"
+}
+
+func (js *JsonSet) checkExpectedType(unmarshalled interface{}) error {
+	if js.ExpectedType == Undefined {
+		return nil
+	}
+	switch unmarshalled.(type) {
+	case bool:
+		if js.ExpectedType != Bool {
+			return fmt.Errorf("Invalid type. Value is a bool. Expected %s", js.expectedType())
+		}
+	case float64:
+		if js.ExpectedType != Number {
+			return fmt.Errorf("Invalid type. Value is a number. Expected %s", js.expectedType())
+		}
+	case string:
+		if js.ExpectedType != String {
+			return fmt.Errorf("Invalid type. Value is a string. Expected %s", js.expectedType())
+		}
+	case []interface{}:
+		if js.ExpectedType != Array {
+			return fmt.Errorf("Invalid type. Value is an array. Expected %s", js.expectedType())
+		}
+	case map[string]interface{}:
+		if js.ExpectedType != Object {
+			return fmt.Errorf("Invalid type. Value is an object. Expected %s", js.expectedType())
+		}
+	}
+
+	return nil
```

Comment:
You always have to return _something_, right? The Go compiler has yelled at me before if I don't include this line. Maybe it was in a different scenario?